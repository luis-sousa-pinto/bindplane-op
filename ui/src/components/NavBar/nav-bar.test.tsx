import { render, screen } from "@testing-library/react";
import { NavBar } from ".";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

describe("NavBar", () => {
  it("main navigation", () => {
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );

    const overviewButton = screen.getByText("Overview");
    expect(overviewButton).toBeInTheDocument();
    expect(overviewButton).toHaveAttribute("href", "/overview");

    const agentsButton = screen.getByText("Agents");
    expect(agentsButton).toBeInTheDocument();
    expect(agentsButton).toHaveAttribute("href", "/agents");

    const configurationsButton = screen.getByText("Configurations");
    expect(configurationsButton).toBeInTheDocument();
    expect(configurationsButton).toHaveAttribute("href", "/configurations");
  });

  it("sub navigation", () => {
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );

    const docLink = screen.getByTestId("doc-link");
    expect(docLink).toHaveAttribute(
      "href",
      "https://docs.bindplane.observiq.com/docs"
    );

    const supportLink = screen.getByTestId("support-link");
    expect(supportLink).toHaveAttribute(
      "href",
      "mailto:support.bindplaneop@observiq.com"
    );

    const slackLink = screen.getByTestId("slack-link");
    expect(slackLink).toHaveAttribute(
      "href",
      "https://observiq.com/support-bindplaneop/"
    );
  });

  it("displays settings menu", () => {
    render(
      <MemoryRouter>
        <NavBar />
      </MemoryRouter>
    );

    const settingsButton = screen.getByTestId("settings-button");
    expect(settingsButton).toBeInTheDocument();

    settingsButton.click();

    const logoutButton = screen.getByText("Logout");
    expect(logoutButton).toBeInTheDocument();
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <NavBar />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
