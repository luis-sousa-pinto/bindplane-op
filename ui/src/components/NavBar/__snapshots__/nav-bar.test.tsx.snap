// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root app-bar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root toolbar MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <svg
        aria-label="bindplane-logo"
        className="logo"
        id="a"
        viewBox="0 0 1249 317"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="317"
          width="1054"
          x="-4"
          y="18"
        />
        <rect
          fill="#fff"
          height="192"
          rx="25"
          ry="25"
          width="192"
          x="1053.02249"
          y="46"
        />
        <path
          d="M1149.40756,113.455c0,8.231-1.5,15.30699-4.51001,21.23-3.01001,5.897-7.14001,10.439-12.39001,13.62601-5.21997,3.162-11.15002,4.743-17.80005,4.743-6.67993,0-12.64001-1.58099-17.88-4.743-5.21997-3.18701-9.33997-7.742-12.34998-13.664s-4.51001-12.98601-4.51001-21.192c0-8.232,1.5-15.29601,4.51001-21.193,3.01001-5.9223,7.13-10.4644,12.34998-13.6263,5.23999-3.1871,11.20007-4.78059,17.88-4.78059,6.65002,0,12.58008,1.5935,17.80005,4.78059,5.25,3.1619,9.38,7.704,12.39001,13.6263,3.01001,5.897,4.51001,12.961,4.51001,21.193Zm-11.51001,0c0-6.274-1.02002-11.557-3.05005-15.848-2.01001-4.316-4.77002-7.578-8.28003-9.78671-3.48999-2.2334-7.43994-3.3501-11.85999-3.3501-4.43994,0-8.41003,1.1167-11.90002,3.3501-3.47998,2.2087-6.23999,5.4707-8.27991,9.78671-2.01001,4.291-3.01001,9.574-3.01001,15.848,0,6.27299,1,11.56799,3.01001,15.884,2.03992,4.29199,4.79993,7.554,8.27991,9.787,3.48999,2.209,7.46008,3.31299,11.90002,3.31299,4.42004,0,8.37-1.10399,11.85999-3.31299,3.51001-2.233,6.27002-5.49501,8.28003-9.787,2.03003-4.31601,3.05005-9.61101,3.05005-15.884Zm26.0199,38.545V74.9091h27.47009c6,0,10.96997,1.0916,14.90991,3.2748,3.94006,2.1833,6.89001,5.1696,8.8501,8.9589,1.94995,3.7642,2.92993,8.0052,2.92993,12.7232,0,4.743-.98999,9.009-2.96997,12.798-1.95996,3.76399-4.92004,6.75099-8.88,8.959-3.93994,2.183-8.90002,3.27499-14.87,3.27499h-18.90002v-9.86199h17.83997c3.79004,0,6.87-.653,9.2301-1.958,2.34998-1.33,4.08997-3.137,5.18994-5.42001,1.09998-2.284,1.66003-4.881,1.66003-7.792s-.56006-5.49599-1.66003-7.754c-1.09998-2.259-2.84998-4.028-5.22998-5.308-2.35999-1.2799-5.46997-1.9198-9.33997-1.9198h-14.6001v67.1158h-11.63Z"
          fill="#69b6e3"
        />
        <g>
          <path
            d="M29.93408,133.74463c8.0332-13.73438,24.87744-20.73145,42.49854-20.73145,33.4292,0,61.15723,24.35938,61.15723,62.97119,0,38.35254-28.24609,63.23047-61.41602,63.23047-16.84424,0-34.20654-7.51562-42.23975-21.25v8.0332c0,6.47852-5.18311,11.9209-11.92041,11.9209-6.47852,0-11.66162-5.44238-11.66162-11.9209V60.40771c0-6.47852,5.18311-11.92041,11.66162-11.92041,6.7373,0,11.92041,5.44189,11.92041,11.92041v73.33691Zm40.42578,82.40674c21.76758,0,39.12988-16.58496,39.12988-40.42676,0-24.09961-17.3623-39.64795-39.12988-39.64795-19.43555,0-40.16699,13.73438-40.16699,39.64795,0,25.65527,18.6582,40.42676,40.16699,40.42676Z"
            fill="#5a595c"
          />
          <path
            d="M160.01904,56.00244c9.06982,0,16.32568,6.47852,16.32568,15.28906,0,8.81104-7.25586,15.28955-16.32568,15.28955-8.81055,0-16.32568-6.47852-16.32568-15.28955,0-8.81055,7.51514-15.28906,16.32568-15.28906Zm11.92041,70.22705c0-6.47852-5.18262-11.92041-11.92041-11.92041-6.47852,0-11.66113,5.44189-11.66113,11.92041v99.76855c0,6.47852,5.18262,11.9209,11.66113,11.9209,6.73779,0,11.92041-5.44238,11.92041-11.9209V126.22949Z"
            fill="#5a595c"
          />
          <path
            d="M217.28516,133.74463c7.25586-14.51221,24.61816-20.73145,37.83447-20.73145,30.06006,0,46.64502,19.69482,46.64502,53.64209v59.34277c0,6.47852-5.18262,11.9209-11.92041,11.9209-6.47852,0-11.66113-5.44238-11.66113-11.9209v-57.78809c0-20.21289-10.10645-32.1333-26.69141-32.1333-17.10303,0-34.20654,9.58838-34.20654,35.24268v54.67871c0,6.47852-5.18262,11.9209-11.92041,11.9209-6.47852,0-11.66113-5.44238-11.66113-11.9209V126.22949c0-6.47852,5.18262-11.92041,11.66113-11.92041,6.73779,0,11.92041,5.44189,11.92041,11.92041v7.51514Z"
            fill="#5a595c"
          />
          <path
            d="M419.15381,60.40771c0-6.47852,5.18262-11.92041,11.92041-11.92041,6.47852,0,11.66113,5.44189,11.66113,11.92041V225.99805c0,6.47852-5.18262,11.9209-11.66113,11.9209-6.73779,0-11.92041-5.44238-11.92041-11.9209v-8.0332c-8.0332,13.73438-25.396,21.25-42.23975,21.25-33.16992,0-61.4165-24.87793-61.4165-63.23047,0-38.61182,27.72803-62.97119,61.15723-62.97119,17.62158,0,34.46582,6.99707,42.49902,20.73145V60.40771Zm-40.42578,75.66895c-21.76807,0-39.13037,15.54834-39.13037,39.64795,0,23.8418,17.3623,40.42676,39.13037,40.42676,21.5083,0,40.1665-14.77148,40.1665-40.42676,0-25.91357-20.73096-39.64795-40.1665-39.64795Z"
            fill="#5a595c"
          />
          <path
            d="M488.85938,284.30469c0,6.47852-5.18262,11.9209-11.92041,11.9209-6.47852,0-11.66162-5.44238-11.66162-11.9209V126.22949c0-6.47852,5.18311-11.92041,11.66162-11.92041,6.73779,0,11.92041,5.44189,11.92041,11.92041v8.29248c8.0332-13.73438,25.39551-21.50879,42.23975-21.50879,33.16992,0,60.63867,25.13672,60.63867,63.48877s-27.46875,62.71289-60.63867,62.71289c-16.84424,0-34.20654-6.73828-42.23975-20.73145v65.82129Zm39.90771-68.15332c21.76758,0,39.12988-15.29004,39.12988-39.38965s-17.3623-40.68506-39.12988-40.68506c-21.24951,0-39.64844,15.03027-39.64844,40.68506,0,25.91406,20.21289,39.38965,39.64844,39.38965Z"
            fill="#fff"
          />
          <path
            d="M630.34668,225.99805c0,6.47852-5.18262,11.9209-11.91992,11.9209-6.47852,0-11.66162-5.44238-11.66162-11.9209V60.40771c0-6.47852,5.18311-11.92041,11.66162-11.92041,6.7373,0,11.91992,5.44189,11.91992,11.92041V225.99805Z"
            fill="#fff"
          />
          <path
            d="M749.29199,217.96484c-8.0332,13.73438-25.6543,21.25-42.49902,21.25-33.16992,0-61.41602-24.87793-61.41602-63.23047,0-38.61182,28.24609-62.97119,61.41602-62.97119,16.84473,0,34.46582,6.99707,42.49902,20.73145v-7.51514c0-6.47852,5.18262-11.92041,11.66113-11.92041,6.73828,0,11.9209,5.44189,11.9209,11.92041v99.76855c0,6.47852-5.18262,11.9209-11.9209,11.9209-6.47852,0-11.66113-5.44238-11.66113-11.9209v-8.0332Zm-40.42578-81.88818c-21.76758,0-39.38867,15.54834-39.38867,39.64795,0,23.8418,17.62109,40.42676,39.38867,40.42676,21.50879,0,40.16699-14.77148,40.16699-40.42676,0-25.91357-20.73145-39.64795-40.16699-39.64795Z"
            fill="#fff"
          />
          <path
            d="M818.22363,133.74463c7.25586-14.51221,24.61816-20.73145,37.83398-20.73145,30.06055,0,46.64551,19.69482,46.64551,53.64209v59.34277c0,6.47852-5.18262,11.9209-11.9209,11.9209-6.47852,0-11.66113-5.44238-11.66113-11.9209v-57.78809c0-20.21289-10.10645-32.1333-26.69141-32.1333-17.10352,0-34.20605,9.58838-34.20605,35.24268v54.67871c0,6.47852-5.18359,11.9209-11.9209,11.9209-6.47852,0-11.66113-5.44238-11.66113-11.9209V126.22949c0-6.47852,5.18262-11.92041,11.66113-11.92041,6.7373,0,11.9209,5.44189,11.9209,11.92041v7.51514Z"
            fill="#fff"
          />
          <path
            d="M1029.16016,185.05371h-89.14355c2.85059,20.21387,18.6582,32.91113,39.12988,32.91113,14.51172,0,22.54492-3.62793,30.83789-12.69727,4.40527-4.92383,11.14258-6.47852,15.80762-3.11035,4.66406,3.62793,5.70117,10.625,2.07324,14.77148-11.66113,15.03027-29.80176,22.28613-49.49609,22.28613-34.46582,0-61.93457-25.13672-61.93457-63.49023s27.46875-62.71143,61.93457-62.71143,60.12012,24.35938,60.12012,62.71143c0,5.44238-4.40527,9.3291-9.3291,9.3291Zm-13.99316-18.39844c-2.85059-20.21289-17.3623-32.39258-36.79785-32.39258-19.9541,0-35.50195,12.17969-38.35254,32.39258h75.15039Z"
            fill="#fff"
          />
        </g>
      </svg>
    </a>
    <div
      className="main-nav"
    >
      <a
        className="nav-link"
        href="/overview"
        onClick={[Function]}
      >
        <svg
          className="icon"
          fill="currentColor"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          viewBox="-1 -1 28 28"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3.69231 0C1.66226 0 0 1.66226 0 3.69231C0 5.72236 1.66226 7.38462 3.69231 7.38462C4.34495 7.38462 4.94712 7.19712 5.48077 6.89423L7.5 9.14423C6.86178 9.9375 6.46154 10.9147 6.46154 12C6.46154 13.006 6.80048 13.9219 7.35577 14.6827L5.74038 16.3269C5.73317 16.3233 5.71875 16.3305 5.71154 16.3269C5.1274 15.9375 4.44231 15.6923 3.69231 15.6923C1.66226 15.6923 0 17.3546 0 19.3846C0 21.4147 1.66226 23.0769 3.69231 23.0769C5.72236 23.0769 7.38462 21.4147 7.38462 19.3846C7.38462 18.7788 7.21514 18.2163 6.95192 17.7115L8.71154 15.9519C9.40745 16.3738 10.2115 16.6154 11.0769 16.6154C11.7332 16.6154 12.357 16.4639 12.9231 16.2115L13.9904 17.7404C13.3377 18.4075 12.9231 19.3089 12.9231 20.3077C12.9231 22.3377 14.5853 24 16.6154 24C18.6454 24 20.3077 22.3377 20.3077 20.3077C20.3077 18.2776 18.6454 16.6154 16.6154 16.6154C16.2548 16.6154 15.905 16.6911 15.5769 16.7885L14.4231 15.1442C15 14.5313 15.4291 13.774 15.6058 12.9231H16.7596C17.1707 14.5096 18.5986 15.6923 20.3077 15.6923C22.3377 15.6923 24 14.03 24 12C24 9.96995 22.3377 8.30769 20.3077 8.30769C18.5986 8.30769 17.1707 9.49038 16.7596 11.0769H15.6058C15.3858 10.0168 14.7764 9.08654 13.9615 8.42308L14.3654 7.32692C14.4988 7.34135 14.6322 7.38462 14.7692 7.38462C16.7993 7.38462 18.4615 5.72236 18.4615 3.69231C18.4615 1.66226 16.7993 0 14.7692 0C12.7392 0 11.0769 1.66226 11.0769 3.69231C11.0769 4.91827 11.7079 5.98918 12.6346 6.66346L12.2885 7.55769C11.899 7.44952 11.4988 7.38462 11.0769 7.38462C10.2909 7.38462 9.56611 7.60817 8.91346 7.96154L6.80769 5.65385C7.17188 5.08414 7.38462 4.41346 7.38462 3.69231C7.38462 1.66226 5.72236 0 3.69231 0ZM3.69231 1.84615C4.72356 1.84615 5.53846 2.66106 5.53846 3.69231C5.53846 4.20793 5.32212 4.65865 4.99038 4.99038C4.65865 5.32212 4.20793 5.53846 3.69231 5.53846C2.66106 5.53846 1.84615 4.72356 1.84615 3.69231C1.84615 2.66106 2.66106 1.84615 3.69231 1.84615ZM14.7692 1.84615C15.8005 1.84615 16.6154 2.66106 16.6154 3.69231C16.6154 4.72356 15.8005 5.53846 14.7692 5.53846C13.738 5.53846 12.9231 4.72356 12.9231 3.69231C12.9231 2.66106 13.738 1.84615 14.7692 1.84615ZM11.0769 9.23077C12.6058 9.23077 13.8462 10.4712 13.8462 12C13.8425 12.0397 13.8425 12.0757 13.8462 12.1154C13.7849 13.5865 12.5661 14.7692 11.0769 14.7692C10.3846 14.7692 9.74639 14.524 9.25962 14.1058C8.67188 13.5974 8.30769 12.8365 8.30769 12C8.30769 10.4712 9.54808 9.23077 11.0769 9.23077ZM20.3077 10.1538C21.3389 10.1538 22.1538 10.9688 22.1538 12C22.1538 13.0313 21.3389 13.8462 20.3077 13.8462C19.2764 13.8462 18.4615 13.0313 18.4615 12C18.4615 10.9688 19.2764 10.1538 20.3077 10.1538ZM3.69231 17.5385C4.72356 17.5385 5.53846 18.3534 5.53846 19.3846C5.53846 20.4159 4.72356 21.2308 3.69231 21.2308C2.66106 21.2308 1.84615 20.4159 1.84615 19.3846C1.84615 18.3534 2.66106 17.5385 3.69231 17.5385ZM16.6154 18.4615C17.6466 18.4615 18.4615 19.2764 18.4615 20.3077C18.4615 21.3389 17.6466 22.1538 16.6154 22.1538C15.5841 22.1538 14.7692 21.3389 14.7692 20.3077C14.7692 19.6947 15.0685 19.1719 15.5192 18.8365C15.6599 18.7897 15.7897 18.7103 15.8942 18.6058C15.905 18.6022 15.9123 18.5805 15.9231 18.5769C16.1358 18.494 16.3702 18.4615 16.6154 18.4615Z"
          />
        </svg>
        Overview
      </a>
      <a
        className="nav-link"
        href="/agents"
        onClick={[Function]}
      >
        <svg
          className="icon"
          fill="none"
          height="20"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="20"
        >
          <rect
            height="7"
            width="7"
            x="3"
            y="3"
          />
          <rect
            height="7"
            width="7"
            x="14"
            y="3"
          />
          <rect
            height="7"
            width="7"
            x="14"
            y="14"
          />
          <rect
            height="7"
            width="7"
            x="3"
            y="14"
          />
        </svg>
        Agents
      </a>
      <a
        className="nav-link"
        href="/configurations"
        onClick={[Function]}
      >
        <svg
          className="icon"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="4"
            x2="4"
            y1="21"
            y2="14"
          />
          <line
            x1="4"
            x2="4"
            y1="10"
            y2="3"
          />
          <line
            x1="12"
            x2="12"
            y1="21"
            y2="12"
          />
          <line
            x1="12"
            x2="12"
            y1="8"
            y2="3"
          />
          <line
            x1="20"
            x2="20"
            y1="21"
            y2="16"
          />
          <line
            x1="20"
            x2="20"
            y1="12"
            y2="3"
          />
          <line
            x1="1"
            x2="7"
            y1="14"
            y2="14"
          />
          <line
            x1="9"
            x2="15"
            y1="8"
            y2="8"
          />
          <line
            x1="17"
            x2="23"
            y1="16"
            y2="16"
          />
        </svg>
        Configurations
      </a>
      <a
        className="nav-link"
        href="/destinations"
        onClick={[Function]}
      >
        <svg
          className="icon destination-icon"
          fill="currentColor"
          height="20"
          stroke="currentColor"
          viewBox="0 -1 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M23.4646 0.557089C23.1153 0.202015 22.6486 0.00419615 22.1505 6.66056e-05L22.1349 0C21.1153 0 20.2788 0.829639 20.2704 1.84937C20.2664 2.34745 20.4564 2.81735 20.8058 3.17243C21.1551 3.5275 21.6217 3.72532 22.1198 3.72945L22.1355 3.72951C23.1551 3.72951 23.9915 2.89988 23.9999 1.88014C24.004 1.38207 23.8139 0.912164 23.4646 0.557089Z"
          />
          <path
            d="M1.88014 10.1353L1.86449 10.1352C0.844823 10.1352 0.00845643 10.9649 6.41193e-05 11.9847C-0.00839479 13.0128 0.821178 13.8562 1.84937 13.8648H1.86495C2.88462 13.8648 3.72105 13.0351 3.72945 12.0154C3.7379 10.9872 2.90833 10.1438 1.88014 10.1353Z"
          />
          <path
            d="M8.89748 3.11794L8.88182 3.11787C7.86216 3.11787 7.02572 3.94751 7.0174 4.96725C7.00894 5.99544 7.83851 6.83886 8.8667 6.84732L8.88236 6.84739C9.90195 6.84739 10.7384 6.01775 10.7468 4.99802C10.7552 3.96976 9.92567 3.12633 8.89748 3.11794Z"
          />
          <path
            d="M13.9651 8.18556L13.9494 8.18549C12.9298 8.18549 12.0934 9.01513 12.085 10.0349C12.0766 11.0631 12.9062 11.9065 13.9344 11.9149L13.95 11.915C14.9696 11.915 15.806 11.0854 15.8144 10.0656C15.8229 9.03738 14.9932 8.19402 13.9651 8.18556Z"
          />
          <path
            d="M20.8068 2.68811L14.7791 8.7312L15.2668 9.2177L21.2945 3.17461L20.8068 2.68811Z"
          />
          <path
            d="M10.2006 5.8146L9.7135 6.30173L12.6332 9.22145L13.1203 8.73433L10.2006 5.8146Z"
          />
          <path
            d="M7.56358 5.81404L2.69617 10.6814L3.18329 11.1686L8.0507 6.30116L7.56358 5.81404Z"
          />
          <path
            d="M20.7637 7.0947C20.2114 7.0947 19.7637 7.54241 19.7637 8.0947V23C19.7637 23.5523 20.2114 24 20.7637 24H22.9999C23.5522 24 23.9999 23.5523 23.9999 23V8.0947C23.9999 7.54241 23.5522 7.0947 22.9999 7.0947H20.7637Z"
          />
          <path
            d="M14.1758 14.1893C13.6235 14.1893 13.1758 14.637 13.1758 15.1893V23C13.1758 23.5523 13.6235 24 14.1758 24H16.412C16.9643 24 17.412 23.5523 17.412 23V15.1893C17.412 14.637 16.9643 14.1893 16.412 14.1893H14.1758Z"
          />
          <path
            d="M7.58777 13.6826C7.03548 13.6826 6.58777 14.1303 6.58777 14.6826V23C6.58777 23.5523 7.03548 24 7.58777 24H9.82402C10.3763 24 10.824 23.5523 10.824 23V14.6826C10.824 14.1303 10.3763 13.6826 9.82402 13.6826H7.58777Z"
          />
          <path
            d="M1 16.2164C0.447716 16.2164 0 16.6641 0 17.2164V23C0 23.5523 0.447715 24 1 24H3.23626C3.78854 24 4.23625 23.5523 4.23626 23L4.23631 17.2164C4.23631 16.6641 3.7886 16.2164 3.23631 16.2164H1Z"
          />
        </svg>
        Destinations
      </a>
    </div>
    <div
      className="sub-nav"
    >
      <a
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium button css-zylse7-MuiButtonBase-root-MuiIconButton-root"
        data-testid="doc-link"
        href="https://docs.bindplane.observiq.com/docs"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        target="_blank"
      >
        <svg
          className="icon"
          fill="none"
          height="20"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="20"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
            stroke="currentColor"
          />
          <path
            d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
            stroke="currentColor"
          />
          <line
            stroke="currentColor"
            x1="12"
            x2="12.01"
            y1="17"
            y2="17"
          />
        </svg>
      </a>
      <a
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium button css-zylse7-MuiButtonBase-root-MuiIconButton-root"
        data-testid="support-link"
        href="mailto:support.bindplaneop@observiq.com"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        target="_blank"
      >
        <svg
          className="icon"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
          />
          <polyline
            points="22,6 12,13 2,6"
          />
        </svg>
      </a>
      <a
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium button css-zylse7-MuiButtonBase-root-MuiIconButton-root"
        data-testid="slack-link"
        href="https://observiq.com/support-bindplaneop/"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        target="_blank"
      >
        <svg
          className="icon"
          fill="none"
          height="24"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14.5 10c-.83 0-1.5-.67-1.5-1.5v-5c0-.83.67-1.5 1.5-1.5s1.5.67 1.5 1.5v5c0 .83-.67 1.5-1.5 1.5z"
          />
          <path
            d="M20.5 10H19V8.5c0-.83.67-1.5 1.5-1.5s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
          />
          <path
            d="M9.5 14c.83 0 1.5.67 1.5 1.5v5c0 .83-.67 1.5-1.5 1.5S8 21.33 8 20.5v-5c0-.83.67-1.5 1.5-1.5z"
          />
          <path
            d="M3.5 14H5v1.5c0 .83-.67 1.5-1.5 1.5S2 16.33 2 15.5 2.67 14 3.5 14z"
          />
          <path
            d="M14 14.5c0-.83.67-1.5 1.5-1.5h5c.83 0 1.5.67 1.5 1.5s-.67 1.5-1.5 1.5h-5c-.83 0-1.5-.67-1.5-1.5z"
          />
          <path
            d="M15.5 19H14v1.5c0 .83.67 1.5 1.5 1.5s1.5-.67 1.5-1.5-.67-1.5-1.5-1.5z"
          />
          <path
            d="M10 9.5C10 8.67 9.33 8 8.5 8h-5C2.67 8 2 8.67 2 9.5S2.67 11 3.5 11h5c.83 0 1.5-.67 1.5-1.5z"
          />
          <path
            d="M8.5 5H10V3.5C10 2.67 9.33 2 8.5 2S7 2.67 7 3.5 7.67 5 8.5 5z"
          />
        </svg>
      </a>
      <button
        aria-haspopup="true"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium button css-zylse7-MuiButtonBase-root-MuiIconButton-root"
        data-testid="settings-button"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          className="icon"
          fill="none"
          height="20"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width="20"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </button>
    </div>
  </div>
</header>
`;
